packages/aws-cdk-lib/aws-ec2/test/machine-image.test.ts

```diff
@@ -404,6 +404,35 @@ describe('latest amazon linux', () => {
       },
     ]);
   });
+
+  test('AmazonLinuxImage with AMAZON_LINUX_2023', () => {
+    // WHEN
+    new ec2.AmazonLinuxImage({ generation: ec2.AmazonLinuxGeneration.AMAZON_LINUX_2023 }).getImage(stack);
+
+    // THEN
+    Template.fromStack(stack).hasParameter('*', {
+      Type: 'AWS::SSM::Parameter::Value<AWS::EC2::Image::Id>',
+      Default: '/aws/service/ami-amazon-linux-latest/al2023-ami-kernel-6.1-x86_64',
+    });
+  });
+});
+
+test('throw error if storage param is set for Amazon Linux 2023', () => {
+  expect(() => {
+    new ec2.AmazonLinuxImage({
+      generation: ec2.AmazonLinuxGeneration.AMAZON_LINUX_2023,
+      storage: ec2.AmazonLinuxStorage.GENERAL_PURPOSE,
+    }).getImage(stack);
+  }).toThrow(/Storage parameter does not exist in SSM parameter name for Amazon Linux 2023./);
+});
+
+test('throw error if virtualization param is set for Amazon Linux 2023', () => {
+  expect(() => {
+    new ec2.AmazonLinuxImage({
+      generation: ec2.AmazonLinuxGeneration.AMAZON_LINUX_2023,
+      virtualization: ec2.AmazonLinuxVirt.HVM,
+    }).getImage(stack);
+  }).toThrow(/Virtualization parameter does not exist in SSM parameter name for Amazon Linux 2023./);
```

Comment:
These tests are outside of the describe block. Is that intended? It looks like theres a mix of 2022 and 2023 tests in the latest describe block. Is that intended too?

Reply:
@tam0ri

Reply:
@scanlonp Thank you for your comment!

> These tests are outside of the describe block. Is that intended?

Yes, this is intentional. I added three tests.

1. 'AmazonLinuxImage with AMAZON_LINUX_2023'
2. 'throw error if storage param is set for Amazon Linux 2023'
3. 'throw error if virtualization param is set for Amazon Linux 2023'

1 is located in `latest` describe block. This is because I want to verify whether AmazonLinuxImage construct generates SSM parameter path for the latest AL2023 image by this change. 2 and 3 are not located in the describe block. This is also intentional because these tests are similar with the following tests located outside of the describe block.

https://github.com/aws/aws-cdk/blob/820bb994a508c96641adbb1ad011fc5def621fae/packages/aws-cdk-lib/aws-ec2/test/machine-image.test.ts#L257-L275

If I should move 2 and 3 into directly below the above tests, please let me know.


> It looks like theres a mix of 2022 and 2023 tests in the latest describe block. Is that intended too?

This is not my intention. Before I added these tests, tests for both 2022 and 2023 are located in the describe block as below.
https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-ec2/test/machine-image.test.ts#L300-L407